Tests/NIOTests/EventLoopTest.swift

```diff
@@ -1295,4 +1295,22 @@ public final class EventLoopTest : XCTestCase {
             XCTFail("Not the same")
         }
     }
+ 
+    public func testScheduleMaximum() throws {
+        let eventLoop = EmbeddedEventLoop()
+        let maxAmount: TimeAmount = .nanoseconds(.max)
+        let scheduled = eventLoop.scheduleTask(in: maxAmount) { true }
```

Comment:
Can we add a comment here referencing the issue we're testing and noting that before our fix this would crash?